### DISCUSSION TITLE
Possible to use outside context in middleware?

### DISCUSSION THREAD

**Discussion by anthonyp** [prisma/prisma#6222] [Q&A]
I understand that Prisma middleware can be set up like so:

```
prisma.$use(async (params, next) => {
  // inspect and maybe do something with params
  next(params);
}
```

However, when running a GraphQL API, I want to bring in outside context that is request-level. For example, imagine that each table in the database has a `createdByUserId` column and I want to automatically inject that into each create operation based on the incoming request's user ID.

It's not pragmatic to create a new Prisma Client for each request; database connections would be overwhelmed.

Is there an elegant pattern for handling this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
